pkg/beacon/beacon.go

```diff
@@ -251,14 +251,24 @@ func confirmCurrentRelayRequest(
 			)
 			return
 		} else if i == maxRetries {
-			logger.Errorf(
-				"could not confirm the expected relay request starting block; "+
-					"the most recent one obtained from chain is [%v] and the "+
-					"expected one is [%v]; giving up after [%v] retries",
-				currentRequestStartBlock,
-				expectedRequestStartBlock,
-				maxRetries,
-			)
+			if currentRequestStartBlock == 0 {
```

Comment:
Looks very good, I'd just add a comment explaining that it's usually happening when entry was submitted very fast before this node received an event and/or was able to confirm request ID.

Reply:
Sure, done.